components/authz-service/server/v2/policy.go

```diff
@@ -169,6 +169,8 @@ func (s *policyServer) CreatePolicy(
 	case storage_errors.ErrConflict:
 		return nil, status.Errorf(codes.AlreadyExists,
 			"policy with id %q already exists", req.Id)
+	case storage_errors.ErrRoleMustExistForStatement:
+		return nil, status.Error(codes.InvalidArgument, "a statement contained a role that does not exist")
```

Comment:
🤔 Would be nice if this included the role in question. To do that, we'd have to make `storage_errors.ErrRoleMustExistForStatement` a struct carrying the non-existing role, but I think it might be worth it.